src/pool/pool.service.ts

```diff
@@ -0,0 +1,160 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { Cron } from '@nestjs/schedule';
+import { InjectRepository } from '@nestjs/typeorm';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import ApolloClient from 'apollo-client';
+import { HttpLink } from 'apollo-link-http';
+
+import gql from 'graphql-tag';
+import { Repository } from 'typeorm';
+import fetch from 'cross-fetch';
+
+import { Pool } from './pool.entity';
+import Big from 'big.js';
+
+export const createClient = (uri: string) =>
```

Comment:
Please initialize the Apollo client in the service constructor once. A separate service for Apollo would be preferrable.